web/static/js/settings.js

```diff
@@ -99,14 +112,39 @@ $(function() {
         </div>
 
         <div id="internetradio_settings" class="addl_settings" style="display:none;">
-          <div class="form-group">
-            <label for="new_internetradio_url">Station Audio URL</label>
-            <input type="text" class="form-control" name="url" id="new_internetradio_url" aria-describedby="urlHelp" data-required="true">
-            <small id="urlHelp" class="form-text text-muted">Audio URL must be supported by <a href="https://www.videolan.org/" target="_blank">VLC</a>.</small>
-          </div>
-          <div class="form-group">
-            <label for="new_internetradio_logo">Station Logo</label>
-            <input type="text" class="form-control" name="logo" id="new_internetradio_logo">
+          <ul class="nav nav-tabs" id="inetradio_tab" role="tablist" style="margin-top:4px;">
+            <li class="nav-item">
+              <a class="nav-link active" href="#inetradio-search-name" id="inetradio-search-name-tab" data-toggle="tab" role="tab" aria-controls="inetradio-search-name" aria-selected="true">Search for Stations</a>
+            </li>
+            <li class="nav-item">
+              <a class="nav-link" href="#inetradio-manual" id="inetradio-manual-tab" data-toggle="tab" role="tab" aria-controls="inetradio-manual" aria-selected="false">Manually Add Station</a>
+            </li>
+          </ul>
+
+          <div class="tab-content" id="inetradio_tab_content" style="padding:15px;">
+            <div class="tab-pane fade show active" id="inetradio-search-name" role="tabpanel" aria-labelledby="inetradio-search-name-tab">
+              <div class="form-group">
+                <label for="inetradio-search-name-txt">Search by Station Name</label>
+                <input type="text" class="form-control" name="inetradio-search-name-txt" id="inetradio-search-name-txt" data-required="true">
+              </div>
+
+              <button type="button" class="btn btn-secondary" id="inetradio-search-name-btn">Search Stations</button>
+
+              <div id="inetradio-search_name_results" style="margin-top:15px;"></div>
+
+            </div>
+            <div class="tab-pane fade" id="inetradio-manual" role="tabpanel" aria-labelledby="inetradio-manual-tab">
+              <div class="form-group">
+                <label for="url">Station Audio URL</label>
+                <input type="text" class="form-control" name="url" id="url" aria-describedby="urlHelp" data-required="true">
+                <small id="urlHelp" class="form-text text-muted">Audio URL must be supported by <a href="https://www.videolan.org/" target="_blank">VLC</a>.</small>
+              </div>
+              <div class="form-group">
+                <label for="logo">Station Logo</label>
+                <input type="text" class="form-control" name="logo">
```

Comment:
needs `id="logo"`, although both the label's for= and the input's id= should probably be a bit more unique like `inetradio-logo`.